app/javascript/chat/components/createChatModal.jsx

```diff
@@ -0,0 +1,65 @@
+import { h } from 'preact';
+import { useState } from 'preact/hooks';
+import PropTypes from 'prop-types';
+import { Modal } from '../../crayons/Modal';
+import { createChannel } from '../actions/chat_channel_setting_actions';
+const CreateChatModal = ({
+  toggleModalCreateChannel,
+  handleCreateChannelSuccess,
+}) => {
+  const [channelName, setchannelName] = useState('');
+  const [userNames, setUserNames] = useState('');
+
+  const handleCreateChannel = async (e) => {
+    e.preventDefault();
+    const result = await createChannel(channelName, userNames);
```

Comment:
refactor (blocking): There should be a check for an empty channel name as the initial value of `channelName` is an empty string and the user could also just submit an empty channel name. One way to handle this client-side as well would be to have the button to create the channel disabled until there is a valid channel name.

I'm assuming this is currently handled on the backend in terms of validation @citizen428?

Reply:
Funnily enough I don't see a validation for `channel_name` in `app/models/chat_channel.rb`. If you try to create a chat channel from a Rails console without any attributes, you'll only see errors for channel_type and slug:

```ruby
cc = ChatChannel.create
cc.save
cc.errors_as_sentence
#=> "Channel type can't be blank, Channel type is not included in the list, and Slug can't be blank"
```

So probably this should be added too.

Reply:
@nickytonline & @citizen428 Added the client-side and backend validation for channel name text field. 
client-side:
- button is disabled until you have no chat channel name. 
backend: 
- added a model validation for this field.

Reply:
Thank you @narender2031!

@mstruve I did a quick check with Blazer, we have one chat channel where the channel name is an empty string which would now be invalid. There are no `ChatChannelMemberships` for this channel, so I think it can safely be deleted. I'll add this is a post-deployment task for someone with console access.